### PR TITLE
Add Color Picker

### PR DESCRIPTION
**Pull request by qsomazzi**

### CODE REVIEW

**Comment by plouc** [comment]
@qsomazzi it's not a color picker, it's more a color selector ;)

File: Resources/views/Form/colorpicker.html.twig [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                $('#{{ id }}').select2({
+                    formatResult:    select2FormatColorSelect,
+                    formatSelection: select2FormatColorSelect,
+                    width:           '100%',
+                    escapeMarkup:    function(m) { return m; }
+                });
+            });
+        </script>
+    {% endspaceless %}
+{% endblock sonata_type_color_picker_widget %}
```
Comment (Soullivaneuh) [inline Resources/views/Form/colorpicker.html.twig]: Need newline here.

File: Form/Type/ColorPickerType.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    const COLOR_FUCHSIA   = 'fuchsia';
+    const COLOR_PURPLE    = 'purple';
+    const COLOR_MAROON    = 'maroon';
+    const COLOR_BLACK     = 'black';
+    const COLOR_GRAY      = 'gray';
+
+    protected $translator;
+
+    /**
+     * @param \Symfony\Component\Translation\TranslatorInterface $translator
```
Comment (Soullivaneuh) [inline Form/Type/ColorPickerType.php]: Import class instead of FQN usage.

File: Form/Type/ColorPickerType.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    const COLOR_OLIVE     = 'olive';
+    const COLOR_LIME      = 'lime';
+    const COLOR_ORANGE    = 'orange';
+    const COLOR_FUCHSIA   = 'fuchsia';
+    const COLOR_PURPLE    = 'purple';
+    const COLOR_MAROON    = 'maroon';
+    const COLOR_BLACK     = 'black';
+    const COLOR_GRAY      = 'gray';
+
+    protected $translator;
```
Comment (Soullivaneuh) [inline Form/Type/ColorPickerType.php]: Missing PHP doc comment.

File: Form/Type/ColorPickerType.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ * file that was distributed with this source code.
+ *
+ */
+
+namespace Sonata\CoreBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatorInterface;
+
```
Comment (Soullivaneuh) [inline Form/Type/ColorPickerType.php]: Useless blank line.

File: Form/Type/ColorPickerType.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
+<?php
+/*
```
Comment (Soullivaneuh) [inline Form/Type/ColorPickerType.php]: Need a blank line after php tag.

File: Form/Type/ColorPickerType.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+     *
+     * @todo Remove it when bumping requirements to SF 2.7+
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $this->configureOptions($resolver);
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (Soullivaneuh) [inline Form/Type/ColorPickerType.php]: `inheritDoc` -> `inheritdoc`

**Comment by Soullivaneuh** [comment]
> @qsomazzi it's not a color picker, it's more a color selector ;)

:+1: 

Also:

* CS need fix: https://travis-ci.org/sonata-project/SonataCoreBundle/jobs/76078354
* Please add related tests.

**Comment by qsomazzi** [comment]
@plouc @Soullivaneuh done.
Color list is now from `http://www.w3schools.com/HTML/html_colornames.asp`

File: Color/Color.php [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+    /**
+     * Return the list of colors.
+     *
+     * @return array
+     */
+    public function getColors()
+    {
+        return array(
+            self::COLOR_ALICEBLUE            => $this->translator->trans('aliceblue', array(), 'SonataCoreBundle'),
```
Comment (rande) [inline Color/Color.php]: Remove the translator, there is no need for this

File: Color/Color.php [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * Return the list of colors.
+     *
+     * @return array
+     */
+    public function getColors()
```
Comment (rande) [inline Color/Color.php]: rename to getAll()

File: Color/Color.php [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Handles A list of all HTML colors.
+ *
+ * @author Quentin Somazzi <qsomazzi@ekino.com>
+ */
+class Color
+{
+    const COLOR_ALICEBLUE            = '#F0F8FF';
```
Comment (rande) [inline Color/Color.php]: Remove the COLOR_ prefix

File: Color/Colors.php [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Color;
+
+/**
+ * Handles A list of all HTML colors.
+ *
```
Comment (Soullivaneuh) [inline Color/Colors.php]: Please add:

```php
@link http://www.w3schools.com/HTML/html_colornames.asp
```

File: Form/Type/ColorSelectorType.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        $this->configureOptions($resolver);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(array(
+            'choices'            => Colors::getAll(),
```
Comment (Soullivaneuh) [inline Form/Type/ColorSelectorType.php]: Would be more user friendly to use [`preferred_choices`](http://symfony.com/doc/2.3/reference/forms/types/choice.html#preferred-choices) options with basic colors like just `red`, `blue`, `green`, `yellow`, `black` etc...

What do you think?

Comment (qsomazzi) [inline Form/Type/ColorSelectorType.php]: I think it's a good idea, will add this

File: Color/Colors.php [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+ * @link   http://www.w3schools.com/HTML/html_colornames.asp
```
Comment (Soullivaneuh) [inline Color/Colors.php]: Why multiples spaces ? I don't think doc tags should be aligned.

File: Form/Type/ColorSelectorType.php (hunk 11 above)
Comment (qsomazzi) [inline Form/Type/ColorSelectorType.php]: done @Soullivaneuh

Comment (Soullivaneuh) [inline Form/Type/ColorSelectorType.php]: :+1:

**Comment by Soullivaneuh** [comment]
@rande as a new feature, this should be on new minor version (2.4).

A 2.3 branch is necessary before merge this one IMO.

**Pull request merged**